src/main/kotlin/pt/up/fe/ni/website/backend/model/Event.kt

```diff
@@ -19,7 +22,24 @@ class Event(
     val description: String,
 
     @JsonProperty(required = true)
-    val date: Date,
+    val startDate: Date,
+
+    val endDate: Date?,
```

Comment:
What do you think about adding a validation to check if endDate is after startDate?
We could perhaps create a new annotation for this (annotations/validation).

Reply:
I've now implemented a class annotation that achieves that. I'm not really happy with the error it produces (it uses the whole model as the value), but I don't really know how I could change that.

Reply:
I see... I understand your approach of targeting the class because we're validating two fields. However, this is out of scope of validations and that's why it doesn't seem possible to override the behaviour. A validation error should always be linked with some field (can you imagine handling special cases in the frontend?), this way we have 2 cases:
- The `errors` field is an array of validations errors that can be displayed in some kind of form
- The `error` field is a message referring to the whole request, which can be displayed in a snackbar for example.

In this case, I think we should simplify our lives and do what would be more convenient for the frontend. Imo, this still makes sense as a validation error and it make sense do display it under the `endDate` field. Therefore, I would change the annotation to target a field and use the `endDate` as the value and param

Reply:
I've been searching and I think it's impossible to access the value of the startDate field inside another field's annotation.

I have found a way to change the propertyPath field of the generated violation, but changing the value field seems to be impossible.

![image](https://user-images.githubusercontent.com/14280476/206870188-d1bc06c1-1893-4c3d-a0ee-29b6d4ee5750.png)

I think this is more acceptable but still not perfect :/

Reply:
With the suggestion of @limwa, I played a bit with `@Embeddable` classes, and I managed to achieve this error:
```
{
    "errors": [
        {
            "message": "endDate must be after startDate",
            "param": "dateInterval",
            "value": {
                "startDate": "28-07-2022",
                "endDate": "27-07-2022"
            }
        }
    ]
}
```

I think this is what we want. What do you think?

Reply:
I think that looks much better, it's now implemented. I also tried playing with `@JsonUnwrapped` to get rid of the nesting, but that would mean that the `param` field would be wrong, so I decided against it.

Reply:
I think either way is fine. I don't dislike `@JsonUnwrapped`, we could even think about changing the `param` to `endDate`, maybe it'd help the frontend side of things since the error will be displayed in a form